src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java

```diff
@@ -33,13 +33,22 @@ public void onBlockPlaceEvent(BlockPlaceEvent event) {
             }
             return;
         }
-
         XMaterial material = XMaterial.matchXMaterial(event.getBlock().getType());
         if (!IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), user, PermissionType.BLOCK_PLACE)) {
             event.setCancelled(true);
             player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotPlaceBlocks.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
         }
+
         IslandBlocks islandBlocks = IridiumSkyblock.getInstance().getIslandManager().getIslandBlock(island.get(), material);
+        int limitUpgradeLevel = IridiumSkyblock.getInstance().getIslandManager().getIslandUpgrade(island.get(), "blocklimit").getLevel();
+        int blockLimit = IridiumSkyblock.getInstance().getUpgrades().blockLimitUpgrade.upgrades.get(limitUpgradeLevel).limits.getOrDefault(material,0);
+
+        if (blockLimit > 0 && islandBlocks.getAmount() >= blockLimit) {
+            player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().blockLimitReached
+                    .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix).replace("%limit%", String.valueOf(blockLimit)).replace("%block%", material.name())));
```

Comment:
You should use something better than Material#name(), enum constant names don't look that good. Maybe replace the _ with spaces, make it lowercase and capitalize it

Reply:
Check out what we did with Biome GUI